tests/test_stl_binders.cpp

```diff
@@ -86,13 +86,13 @@ TEST_SUBMODULE(stl_binders, m) {
 
     // test_noncopyable_containers
     py::bind_vector<std::vector<E_nc>>(m, "VectorENC");
-    m.def("get_vnc", &one_to_n<std::vector<E_nc>>, py::return_value_policy::reference);
```

Comment:
Comments from #2746:

- @bstaletic: These rvps look bogus. The return is by value.
- @YannickJadoul: Huh? I thought it returned a pointer to the container?
- @bstaletic: I was just entirely unable to figure out why/how the reference rvp is here. This test was added in #490 (commit 617fbcf). The PR involves returning elements by reference, but I don't see why these factory methods return the container by reference.
- @YannickJadoul: Huh? I thought it returned a pointer to the container?
- @bstaletic: That's what happens when you (read: I) do a review while really really tired. Listen to @YannickJadoul, everyone!
- @YannickJadoul: Hah 😄 No worries, we've all been there.
- @bstaletic: I'm not resolving this discussion, just so it doesn't get collapsed before others take a look.